src/game/client/worldheightmap.h

```diff
@@ -78,25 +87,72 @@ class WorldHeightMap : public RefCountClass, public WorldHeightMapInterfaceClass
-    virtual float Get_Bilinear_Sample_Seismic_Z_Velocity(int x, int y);
+    WorldHeightMap();
+    WorldHeightMap(ChunkInputStream *strm, bool logical_data_only);
 
-    bool Get_Flip_State(int x, int y);
+#ifdef GAME_DLL
+    WorldHeightMap *Hook_Ctor() { return new (this) WorldHeightMap(); }
+    WorldHeightMap *Hook_Ctor2(ChunkInputStream *strm, bool logical_data_only)
+    {
+        return new (this) WorldHeightMap(strm, logical_data_only);
+    }
+    void Hook_Dtor() { WorldHeightMap::~WorldHeightMap(); }
+#endif
+
+    virtual ~WorldHeightMap() override;
+    virtual int Get_Border_Size() override { return m_borderSize; }
+    virtual float Get_Seismic_Z_Velocity(int x, int y) const override;
+    virtual void Set_Seismic_Z_Velocity(int x, int y, float velocity) override;
+    virtual float Get_Bilinear_Sample_Seismic_Z_Velocity(int x, int y) override;
+
+    bool Get_Cliff_State(int x_index, int y_index) const;
+    bool Get_Flip_State(int x, int y) const;
+    bool Get_Seismic_Update_Flag(int x_index, int y_index) const;
+    void Get_Terrain_Color_At(float x, float y, RGBColor *color);
+    Utf8String Get_Terrain_Name_At(float x, float y);
+
+    TextureClass *Get_Alpha_Terrain_Texture();
+    TextureClass *Get_Edge_Terrain_Texture();
+    TextureClass *Get_Flat_Texture(int x, int y, int pixels_per_cell, int cell_width);
+    TextureClass *Get_Terrain_Texture();
+    int Get_Texture_Class(int x_index, int y_index, bool base_class);
+    TXTextureClass Get_Texture_From_Index(int index);
+
+    void Get_Alpha_UV_Data(
+        int x_index, int y_index, float *const u, float *const v, unsigned char *const alpha, bool *flip, bool full_tile);
+    bool Get_Extra_Alpha_UV_Data(
+        int x_index, int y_index, float *const u, float *const v, unsigned char *const alpha, bool *need_flip, bool *cliff);
+    bool Get_UV_Data(int x_index, int y_index, float *const u, float *const v, bool full_tile);
+    void Get_UV_For_Blend(int edge_class, Region2D *range);
+
+    void Clear_Flip_States();
+    bool Is_Cliff_Mapped_Texture(int x, int y);
+
+    bool Set_Draw_Origin(int x_origin, int y_origin);
+    void Set_Flip_State(int x_index, int y_index, bool state);
+    void Set_Seismic_Update_Flag(int x_index, int y_index, bool flag);
+    void Set_Texture_LOD(int lod);
 
     static int Get_Min_Height_Value() { return 0; }
     static int Get_Max_Height_Value() { return 255; }
-    int Get_Width() { return m_width; }
-    int Get_Height() { return m_height; }
+    int Get_X_Extent() { return m_width; }
+    int Get_Y_Extent() { return m_height; }
     int Border_Size() { return m_borderSize; }
     int Get_Draw_Width() { return m_drawWidthX; }
     int Get_Draw_Height() { return m_drawHeightY; }
     int Get_Draw_Origin_X() { return m_drawOriginX; }
     int Get_Draw_Origin_Y() { return m_drawOriginY; }
+    unsigned char *Get_Data_Ptr() { return m_data; }
+    void Set_Draw_Width(int width) { m_drawWidthX = width; }

```

Comment:
these may not be correct in WB we got 00484BE0 00484C20 which clamp it as well, but not 100% that its those functions

Reply:
fixed.